### PR TITLE
Prac 05 feedback

### PR DESCRIPTION
**Pull request by IanEllimanJCU**
Prac 5 code request review
Please check logic and formatting, see if things could be simplified
@Haiqiangwang21

### CODE REVIEW

File: prac_05/hex_colours.py [hunk 1]
```diff
@@ -1,3 +1,20 @@
 COLOUR_CODE = {"maroon": "#b03060", "orchid": "#da70d6", "pink": "#ffc0cb", "salmon": "#fa8072",
                 "thistle": "#d8bfd8", "violet": "#ee82ee", "wheat": "#f5deb3", "linen": "#faf0e6",
-                "lavender": "#e6e6fa", "brown": "#a52a2a"}
\ No newline at end of file
+                "lavender": "#e6e6fa", "brown": "#a52a2a"}
```
Comment (Haiqiangwang21) [inline prac_05/hex_colours.py]: The dict data have repeated.

File: prac_05/hex_colours.py [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+    for i in COLOUR_CODE:
+        if i != "brown":
+            print(f"{i}, ", end='')
+    print("and brown")
+    colour_choice = input("Please enter your colour choice: ").lower()
+    while colour_choice != "":
+        print_colour_code(colour_choice)
+        colour_choice = input("Please enter your colour choice: ").lower()
+    exit()
+
```
Comment (Haiqiangwang21) [inline prac_05/hex_colours.py]: You dont actually need a function for this, you could do a while loop.
eg. 
while colour_name = "":
        if colour_name in colour_code :
           print(
        else
           error

File: prac_05/state_names.py [hunk 3]
```diff
@@ -3,7 +3,6 @@
 State names in a dictionary
 """
```
Comment (Haiqiangwang21) [inline prac_05/state_names.py:4]: The code for this one isnt finished, like you havent add codes prompting user input and etc.

File: prac_05/word_occurrences.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    split_user_string.sort()
+    print(split_user_string)
+    longest_len_word = 0
+    for i in split_user_string:
+        if len(i) > longest_len_word:
+            longest_len_word = len(i)
+        if i in count_word_occurences:
+            count_word_occurences[i] += 1
+        else:
+            count_word_occurences[i] = 1
```
Comment (Haiqiangwang21) [inline prac_05/word_occurrences.py]: there is a short way to count word_length 
eg.
longest_word = max(len(word) for word in words)

File: prac_05/emails.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+def main():
```
Comment (Haiqiangwang21) [inline prac_05/emails.py:1]: Yea it so long man, maybe you could try like one function get the name from the dict and another checking the input and other error etc.
eg. 
def get_name(email):
    parts = email.split("@")[0].split(".")
    name = " ".join(parts).title()
    return name

Comment (IanEllimanJCU) [reply prac_05/emails.py:1]: has been refactored to be simpler in https://github.com/IanEllimanJCU/cp1404practicals/pull/1/commits/00b85e67036b8477b2732c80025965e5f324080a

File: prac_05/state_names.py (hunk 3 above)
Comment (IanEllimanJCU) [reply prac_05/state_names.py:4]: Reviewer didn't see entire code? Seems finished to me.

File: prac_05/hex_colours.py (hunk 1 above)
Comment (IanEllimanJCU) [reply prac_05/hex_colours.py]: Fixed

File: prac_05/hex_colours.py (hunk 2 above)
Comment (IanEllimanJCU) [reply prac_05/hex_colours.py]: implemented feedback https://github.com/IanEllimanJCU/cp1404practicals/commit/e06acba3aa52b11c868df270e67f6b48be56d66b

File: prac_05/word_occurrences.py (hunk 4 above)
Comment (IanEllimanJCU) [reply prac_05/word_occurrences.py]: feedback implemented https://github.com/IanEllimanJCU/cp1404practicals/commit/fbe867decf4ffc690cd2984bb3f3c327c0e5f65f

**Pull request merged**